src/hooks/useTokenList/index.tsx

```diff
@@ -41,10 +42,15 @@ export default function useTokenList(): WrappedTokenInfo[] {
 
 export type TokenMap = { [address: string]: Token }
 
-export function useTokenMap(): TokenMap {
-  const { chainId } = useWeb3React()
+export function useTokenMap(chainId?: SupportedChainId): TokenMap {
+  let { chainId: _chainId } = useWeb3React()
+
+  if (chainId) {
```

Comment:
This can be simpler as a one-line defaulting statement: `chainId = chainId || activeChainId`